### DISCUSSION TITLE
Thank You!

### DISCUSSION THREAD

**Discussion by siebertm** [1coderookie/KobraGoNeoInsights#12] [General]
I just wanted to say thank you for this huge amount of information you put together!

Especially the part about Klipper (also thanks to the original sources), I did the change to klipper+mainsailos and I've never seen a better first layer on my Go! 

Okay, it's my first printer, and I'm learning, so maybe it's not klipper and I just learned a lot, but thanks anyways!

**Comment by 1coderookie** [comment]
Thanks for your kind words, I really appreciate it :+1:   
Please consider starring this repo then as I'm always excited about new stars ;)  

Yes, Klipper is a beauty and with Klipperizing these machines they actually become useable and reliable.  
Enjoy printing and watching some 1st layer p0rn ;)

**Reply by NuclearPhoenixx** [reply]
Yeah thanks a lot from me too. As a beginner it's extremely helpful to have this kind of thorough documentation on how the device works. Also, I like your notes and warnings on things!
(Though, if I'd follow everything that you "highly suggest", I'd almost need to rebuild the whole printer from scratch ;P)

**Reply by 1coderookie** [reply]
Thanks for the feedback! :smiley_cat:  
Yeah, I really like those expandable textboxes (generally speaking) as well, as they allow me to add notes and even longer descriptions of processes or so while keeping the main section as short as possible. I know it's already quite a lot of text, but at least it's not completely overwhelming when using those boxes. ;)

Haha, right, but I really can highly suggest or recommend those. Especially taking it apart and check the frame and stuff is just SO crucial - if the hardware is trammed well, less weird problems will appear. So if you ever have a weekend to tinker at it, maybe take the time to do so and then you can also e.g. insulate the bedplate, switch the PTFE tube and so on. In the end it'll really pay out! ;)

**Reply by NuclearPhoenixx** [reply]
Oh yeah, because of you I've already got a todo list of my own haha! Jokes aside, it's extremely helpful to have that, especially as a beginner, so thanks again.

So far everything worked pretty well considering it's a *really* cheap printer. Did most of the things you recommended that didn't involve buying new parts right away or something. The weirdest thing to me is the fact that they crimped the hotbed wires for the mainboard screw terminal, but only soldered the other 24V connector. Other than that I'm pretty sure my bed is slightly warped, but I'll fix that with some adjustable spacers when I go to insulate it. For now I get really good prints considering the printer and that I'm only in this for a week now. I've learned so much though and I love it :D

**Reply by 1coderookie** [reply]
Hehe nice, so enjoy tinkering then! I also do have a few todo lists, but I actually don't have enough time to work on all of those LOL  

Welcome, I'm happy it's helpful for you :+1:   

Well, it wasn't always that cheap tho - in the beginning it was about 250! 
But nowadays where they throw them out of their shops cuz of the new Kobra 2 Neo, you can't get more hardware for your money. I mean, just look at what you get in terms of PSU, frame, head, mobo, bed, plate, steppers and so on. That's why I often recommend to people to look out for cheap refurbished or maybe even used ones, as you'll just have a complete set of spare parts for only little money then.  
In the beginning when I got the 1st Neo which was broken (and my 1st 3d printer ever), I had to replace quite a few parts. As the printers weren't that cheap then, I had to buy parts individually - which really summed up.. About half a year later or so I then found refurbished ones on ebay and got myelf two more - with an ebay coupon I got them for 80€ each - what a deal! One is completely disassembled in my drawers, just to have spare parts available if I need it. So for 80€ you'll probably just get the PSU and a new mobo, that's it. Now I have all the parts available tho ;)   

And looking at the new K2Neo and comparing it to the Neo1, I still recommend it to people if they ask me what to buy. The reason is simple: besides the different printhead there's no real improvement imho. And that one (as it's capable of a higher flowrate) can be made up by using a CHT nozzle.  
Z-offset sensor? Well, yeah, might be nice if it would work reliably and if one absolutely doesn't know how to tune in the z-offset, but you have to tune it anyway. So that (for me) isn't really an improvement. Instead, there's no z-limit switch anymore - which I really don't like as the proximity sensor is more prone to fail than a simple optical limit switch imho.  
New mobo? Well, imho there's no advantage of that one either compared to the one from the Neo1, and at the Neo1 you can easily flash Klipper (which I'd really strongly recommend to do ;) ). Using Klipper actually turned my Neo in a reliable machine, before it was just a PITA with that buhhy stock fw..  

Ah, one last note on the adjustable spacers: if you have a *warped* bed, like an inconsistent surface, then adjustable spacers don't really help you anyway. They're more useable when it comes down to equalize certain deviations like a tilted bed. But, again, most of these issues can then be made up with a well trammed machine.  
Besides that, adjusting the bed with those adjustable spacers can be a real PITA as well, especially cuz you have to access them from above (when using thethreads in the gantry instead of drilling them out).  
So if you're really suffering from a severly *warped* bed which the ABL can't take 100% care of, then I'd suggest to try the Kapton tape method first to equalize the whole surface.  
Also keep in mind that the y-offset for the ABL sensor is set wrong in the stock firmware. Which btw I still have to add as an explicit section and write a possible workaround for that as well - that's still on one of my ToDo lists as well.. ;)  
But also here, again: just flash Klipper and you'll end up being happy ;)

**Reply by NuclearPhoenixx** [reply]
Wow, 80€ sounds like a real steal for the printer, that's ridiculously cheap!

Ah well, it's really not that bad, I just noticed some small(-ish) inconsistencies with the layer adhesion if I print on the four outer corners. So probably it's not warped and just a bit tilted and I only need to tram the bed better. I haven't investigated that much tbh, cause like I said I'm already pretty happy with the results and the stock "solution" to tram the bed is a joke (pain) LOL

> Also keep in mind that the y-offset for the ABL sensor is set wrong in the stock firmware.

That's good to know! I'll probably go ahead and compile my own Marlin FW for now and double-check the whole config. Or maybe I'll use that one for now: https://github.com/jokubasver/Kobra_Neo I'll learn a sh*t ton from that as well I'm sure.
Octoprint and Klipper are definitely also on the todo list, but I don't want to get too involved with other devices attached to the printer for now.

**Reply by 1coderookie** [reply]
I'll add that section asap, as you can change that setting using a g-code command instead of recompiling the whole fw.  
Compiling the Marlin fw might be kinda difficult, so be warned. Better use the one from jokubasver then if you wanna stick to the stock fw. He didn't change the probe's offset tho, so that's still not correct there, I'll probably open a PR about it soon..  
It's not *that* much of an issue, but why not being as precise with the probes offset coordinates as possible, right? ;)

**Reply by NuclearPhoenixx** [reply]
> It's not that much of an issue, but why not being as precise with the probes offset coordinates as possible, right? ;)

If I understand it correctly, then it doesn't make a big difference at all, does it? It'll just increase the z-offset by a constant factor, but you'll have to adjust that anyways so meh 🤷🏻

As for the Marlin FW, I'll see if I'll regret that haha. I found some great tutorials, but if it's too much of a pain then I'll definitely check jokubasver's FW out!

**Reply by 1coderookie** [reply]
No, I'm talking about the Y offset! Sorry I forgot to mention that.. So right now the fw thinks the nozzle and the probe are in the same y-location, like being mounted absolutely parallel next to each other. The probe is mounted more to the back tho. So when it compensates while printing, it's actually compensating for the wrong location and not for where the nozzle actually is!

**Reply by NuclearPhoenixx** [reply]
Oh, I see. Well that's pretty stupid...

**Reply by 1coderookie** [reply]
Haha yes, exactly ;)  
I just opened a PR at jokubasver tho..

**Reply by 1coderookie** [reply]
..just quickly added that section to the belonging chapter as well: https://1coderookie.github.io/KobraGoNeoInsights/firmware/fw_marlin/#probes-offset

**Reply by NuclearPhoenixx** [reply]
Cool, thanks

**Reply by NuclearPhoenixx** [reply]
It actually was surprisingly easy to compile the firmware once the keil licensing page didn't HTML500 me. I've created a fork from https://github.com/jokubasver/Kobra_Neo and did some minor mods (e.g. default homing speed was decreased a little bit, was too aggressive for me; changed black background to dark; unlocked more languages, though the translations can be pretty bad and I also don't know how to get the encoding for special letters correct right now, SD card file names are now scrolled, ...). I also reverted the PID and e-steps values back to default so that it can be a drop-in-replacement for anybody. Calibration is necessary anyways of course, but at least it's not based off of one user's values.

https://github.com/NuclearPhoenixx/Kobra_Neo

Any more ideas on what can be changed? I know you're not using Marlin, but maybe you're aware of any other annoying restrictions ;)

**Reply by 1coderookie** [reply]
Nice! I guess I'll link to that one as well then :)  

Hmm, I don't know exactly what @jokubasver already changed as I didn't really take the time to compare everything. Iirc, then he already set the max speed tor the extruder to 50mm/s (https://1coderookie.github.io/KobraGoNeoInsights/firmware/fw_marlin/#limited-retraction-speed_1) - if not, then that would be sth I'd suggest to do in that fw right away as well.

**Reply by NuclearPhoenixx** [reply]
Yup, the default max feedrate for E0 was already set to 50 in the mod, I didn't change that 👍🏻

**Reply by NuclearPhoenixx** [reply]
I also managed to speed up the boot sequence by a factor of 2 by removing some stupid artificial delays and some random startup tune that doesn't work anyways on that machine. Now it only takes about 5 seconds to boot, which is pretty nicer tbh
